### PR TITLE
symbols: fix conflicts with aom

### LINKED ISSUE #453: SVT-AV1 and libaom have conflicting symbols
Not tested by myself (but I'm not surprised if this is true), brought up here: http://ffmpeg.org/pipermail/ffmpeg-devel/2019-July/246504.html

> e.g : SVT-AV1 reused libaom. If you enable both libsvt-av1 and libaom as static link, then will have name conflict. 
  Such issue is not easily found if you keep wrapper patch on https://github.com/OpenVisualCloud/SVT-AV1/tree/master/ffmpeg_plugin ,
  But will easily found if works on FFmpeg mainline.

### PR DESCRIPTION
**Pull request by xuguangxin**
We use many functions from liboam. We will see many duplicate define errors
if we enable liboam static library + svtav1.
this will fix #453

@jsunintel , @hassount @kylophone 
please help review this meet the requirement or not
thanks

### CODE REVIEW

**Comment by DesWurstes** [comment]
The symbols in [tests](https://github.com/OpenVisualCloud/SVT-AV1/tree/master/test) directory needs to be replaced too.

**Comment by kylophone** [comment]
lgtm, can you fix the tests too?

**Comment by xuguangxin** [comment]
@DesWurstes , @kylophone  thanks for point out.  I will work on this.
@hassount  Do you know where to find the definition of [aom_dc_128_predictor_16x16_sse2](https://github.com/OpenVisualCloud/SVT-AV1/search?q=aom_dc_128_predictor_16x16_sse2&unscoped_q=aom_dc_128_predictor_16x16_sse2&type=Code)?

thanks

**Comment by hassount** [comment]
> @DesWurstes , @kylophone thanks for point out. I will work on this.
> @hassount Do you know where to find the definition of [aom_dc_128_predictor_16x16_sse2](https://github.com/OpenVisualCloud/SVT-AV1/search?q=aom_dc_128_predictor_16x16_sse2&unscoped_q=aom_dc_128_predictor_16x16_sse2&type=Code)?
> 
> thanks

Yes, it's an assembly function:
https://github.com/OpenVisualCloud/SVT-AV1/blob/962bd31d978a6eb08d13eb3bf67bd07ffa2e2579/Source/Lib/Common/ASM_SSE2/intrapred_sse2.asm#L323

**Comment by xuguangxin** [comment]
@hassount ,thanks for the information.
I want change aom_dc_128_predictor_16x16_sse2 to eb_aom_dc_128_predictor_16x16_sse2. 
so I changed aom to eb_aom in this line https://github.com/OpenVisualCloud/SVT-AV1/blob/97eff0cde161af5ce6bcece2e6592ce57c659559/Source/Lib/Common/ASM_SSE2/x86inc.asm#L109 
but seems it's still not working, compiler reported
` undefined reference to `eb_aom_dc_predictor_16x16_sse2'
any other tricks?
thanks

**Comment by hassount** [comment]
@xuguangxin 
@1480c1 did something similar in #216  could that help ?

**Comment by xuguangxin** [comment]
no, seems unrelated. @1480c1 any suggestion?
thanks

**Comment by 1480c1** [comment]
The private prefix is a copy over and best to be ignored, just add the prefix manually to the functions, might be better to look at the x86inc.asm and x64inc.asm and try to consolidate them etc.

**Comment by xuguangxin** [comment]
thanks @1480c1 for the suggestion. 
It's related to env, I need to clean my build directory and do cmake from scratch after I change x86.asm.

Now, everything passed. @hassount , please help review it.
I have done the following tests:
1. build with `ffmpeg + libaom ca3651934f9ec2bf70cd855b5cddce9124ef317e`.  There are no conflicts.
2. using `SvtAv1EncApp -i xxx.y4m -b out.ivf  -n 600 -intra-period 119 -q 20`, it can bit match with master.

thanks

**Comment by hassount** [comment]
this LGTM just needs rebasing
@jsunintel, any issues with this ?

**Comment by xuguangxin** [comment]
rebased.
 @jsunintel , any comment on this?
thanks

File: Source/Lib/Common/Codec/EbPsnr.c [hunk 1]
```diff
@@ -166,15 +166,15 @@ static int64_t highbd_get_sse(const uint8_t *a, int32_t a_stride, const uint8_t
     return total_sse;
 }
 
-int64_t aom_get_y_sse_part(const Yv12BufferConfig *a,
+int64_t eb_eb_aom_get_y_sse_part(const Yv12BufferConfig *a,
```
Comment (jsunintel) [inline Source/Lib/Common/Codec/EbPsnr.c]: eb_eb_

**Comment by 1480c1** [comment]
What does eb mean? I've seen it everywhere in the SVT encoders

**Comment by xuguangxin** [comment]
@jsunintel , thanks for point out. You are so careful. 
Yes. there are too many lines, hard change by hand. I use sed.
updated, please help review again.

@1480c1 ,FYI.
https://www.fiercevideo.com/video/intel-netflix-team-up-to-deliver-av1-codec-for-visual-workloads

**Comment by hassount** [comment]
> What does eb mean? I've seen it everywhere in the SVT encoders

eb came in from the initial product / company acquired by Intel called eBrisk video. we chose eb as a prefix to be used in the library.

**Comment by xuguangxin** [comment]
> And why eb? Will svt look better?

@jsunintel , just like other function names. 
If we decided to use svt, maybe we need to change other functions too. 
But, it's not the main purpose of this PR. maybe we can change it later with other function names.
thanks

**Comment by 1480c1** [comment]
Ah, I heard about it, but I thought it was eBrick and could never find anything about it

**Comment by kylophone** [comment]
Thanks! lgtm

**Comment by kylophone** [comment]
Huge diff, could cause a lot merge conflicts. We should still merge it, though. @hassount, are there any big patches coming?

**Comment by hassount** [comment]
> Huge diff, could cause a lot merge conflicts. We should still merge it, though. @hassount, are there any big patches coming?

yes there will be but not any time soon, it would unfair for this not to go through. Let's merge it

**Comment by xuguangxin** [comment]
@kylophone we need to face the pain anyway. Merge it early is better than late. 
@hassount , seems yasm download has some issue. Anyone can help with this?

**Comment by 1480c1** [comment]
> @hassount , seems yasm download has some issue. Anyone can help with this?

It seems <http://www.tortall.net> is down

**Comment by 1480c1** [comment]
https://yasm.tortall.net works, but www doesn't

**Comment by xuguangxin** [comment]
> It seems http://www.tortall.net is down

Wired, I can download http://www.tortall.net/projects/yasm/releases/yasm-1.3.0-win64.exe.

Not sure how many external dependencies we have, maybe we can consider mirror them to a special branch.  We can merge the branch to master on windows.

**Comment by 1480c1** [comment]
As far as I know, direct dependencies are cmake (plus make or whatever generator), a c compiler, and yasm or nasm

I do have a fix for it to use the msys2 provided yasm

**Comment by 1480c1** [comment]
Could you make sure to have a version of yasm downloaded? I still can't access the original tortal.net, when I try to goto tortal.net, it redirects to tortal.com

**Comment by xuguangxin** [comment]
How about use this:
[yasm-1.3.0-win64.txt](https://github.com/OpenVisualCloud/SVT-AV1/files/3420571/yasm-1.3.0-win64.txt)

**Comment by 1480c1** [comment]
That seems like a dumb idea that works

**Comment by xuguangxin** [comment]
> That seems like a dumb idea that works

yeah,  we'd better only depend on github. Other sites may down anytime.

**Comment by 1480c1** [comment]
Although I would say something like it would be better to try to encode that is base64 and the decode in Powershell to make sure that the program is correct, after looking online, it would probably be slightly hard to do that in pure Powershell

**Comment by xuguangxin** [comment]
> A encode that is base64
What is the benefit of base64? Avoid catch by antivirus? 
But it only used by appveyor, a normal user will not download the exe it.

**Comment by xuguangxin** [comment]
@hassount  all check passed.  
Could we merge it?
thanks

**Comment by 1480c1** [comment]
Mainly so that none of the data will get mangled through the download process, but since it seems to work fine, probably ok

**Comment by xuguangxin** [comment]
> but since it seems to work fine, probably ok

@1480c1 do you mind I send other PR to fix the same issue as https://github.com/OpenVisualCloud/SVT-AV1/pull/490

thanks

**Comment by 1480c1** [comment]
Sure, go ahead

**Comment by 1480c1** [comment]
It would be better to see what your PR is and weigh the merits

**Comment by xuguangxin** [comment]
seems tortall.net come back. Now all test passed.
@hassount it's rebased, any objection to merging this?

**Comment by xuguangxin** [comment]
thanks hassount

**Pull request merged**